activity_browser/app/ui/tabs/impact_categories.py

```diff
@@ -45,39 +46,97 @@ def __init__(self, parent):
+        self.mode_radio_list.setToolTip(
+            "List view of impact categories")
+        #
+        self.mode_radio_tree = QtWidgets.QRadioButton("Tree view")
+        self.mode_radio_tree.setToolTip(
+            "Tree view of impact categories\n"
+            "v CML 2001\n"
+            "    v climate change\n"
+            "        CML 2001, climate change, GWP 100a\n"
+            "        ...\n"
+            "You can drag entire 'branches' of impact categories at once")
         #
         search_layout = QtWidgets.QHBoxLayout()
-        search_layout.setAlignment(QtCore.Qt.AlignTop)
-        search_layout.addWidget(header('Impact Categories'))
         search_layout.addWidget(self.search_box)
-        search_layout.addWidget(reset_search_button)
+        search_layout.addWidget(self.reset_search_button)
+        #
+        mode_layout = QtWidgets.QHBoxLayout()
+        mode_layout.setAlignment(QtCore.Qt.AlignTop)
+        mode_layout.addWidget(header('Impact Categories'))
+        search_layout.addWidget(self.mode_radio_list)
+        search_layout.addWidget(self.mode_radio_tree)
+        #
+        mode_layout_container = QtWidgets.QWidget()
+        mode_layout_container.setLayout(mode_layout)
         #
         search_layout_container = QtWidgets.QWidget()
         search_layout_container.setLayout(search_layout)
         #
         container = QtWidgets.QVBoxLayout()
         container.setAlignment(QtCore.Qt.AlignTop)
+        container.addWidget(mode_layout_container)
         container.addWidget(search_layout_container)
         # container.addWidget(horizontal_line())
         container.addWidget(self.table)
+        container.addWidget(self.tree)
+        self.tree.setVisible(False)
         self.setLayout(container)
 
-        reset_search_button.clicked.connect(self.table.sync)
-        reset_search_button.clicked.connect(self.search_box.clear)
+        self.reset_search_button.clicked.connect(self.table.sync)
+        self.reset_search_button.clicked.connect(self.tree.sync)
+
+        self.reset_search_button.clicked.connect(self.search_box.clear)
         self.search_box.returnPressed.connect(lambda: self.table.sync(query=self.search_box.text()))
+        self.search_box.returnPressed.connect(lambda: self.tree.sync(query=self.search_box.text()))
+
         signals.project_selected.connect(self.search_box.clear)
         self.table.new_method.connect(self.method_copied)
 
+        self.connect_signals()
+
+    def connect_signals(self):
+        self.mode_radio_list.toggled.connect(self.button_clicked)
+
     @QtCore.Slot(tuple, name="searchCopiedMethod")
     def method_copied(self, method: tuple) -> None:
         """If a method is successfully copied, sync and filter for new name."""
         query = ", ".join(method)
         self.search_box.setText(query)
         self.table.sync(query)
 
+    @QtCore.Slot(bool, name="isListToggled")
+    def button_clicked(self, toggled: bool):
```

Comment:
Both the `button_clicked` and `update_view` methods can be folded into a single `update_view` with the code below:

```python
@Slot(bool, name="isListToggled")
def update_view(self, toggled: bool):
    self.tree.setVisible(not toggled)
    self.table.setVisible(toggled)
```

Reply:
Sorry about the late reply, I was not watching my GH email (will update that after this comment). I'll update your suggestion and do a new PR :)

Reply:
> 
> 
> Sorry about the late reply, I was not watching my GH email (will update that after this comment). I'll update your suggestion and do a new PR :)

You can also just push a new commit to the current one, that way we keep the comments and feedback :)